kernel: bump 6.12 to 6.12.58

mgz0227 opened the pull request:
changelogs: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.58

removed 
1.target/linux/generic/backport-6.12/612-01-v6.17-net-dsa-tag_brcm-legacy-reorganize-functions.patch
2.target/linux/qualcommax/patches-6.12/0151-arm64-qcom-ipq6018-nss_port5.patch
3.target/linux/realtek/patches-6.12/020-01-v6.18-timer-rtl-otto-work-around-dying-timers.patch
5.target/linux/realtek/patches-6.12/020-03-v6.18-timer-rtl-otto-do-not-interfere-with-interrupts.patch

1.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=a4daaf063f8269a5881154c5b77c5ef6639d65d3

2.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=9a7a5d50ee2e035325de9c720e4842d6759d2374

3.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=d0e217b33d42bfe52ef7ef447916a23a586e6e5c

4.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=8cc561dd9d02f1753ae34dfdd565662828be9a9d

Additional changes:
- Manually adapted bcm27xx patch:
  * 950-0410-media-i2c-adv7180-Add-support-for-V4L2_CID_LINK_FREQ.patch
    Rebased and adjusted for kernel 6.12 to fix context conflicts.
- Synced lantiq DTS (danube.dtsi) with upstream bindings
  to fix DT validation issues on kernel 6.12.
- Manually adapted DTS to match OpenWrt's lantiq DTS layout.
Compile-tested on x86_64
Run-tested on x86_64

mgz0227 committed:
kernel: bump 6.12 to 6.12.58

changelogs:
https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.58

Removed upstreamed patches:
1. target/linux/generic/backport-6.12/612-01-v6.17-net-dsa-tag_brcm-legacy-reorganize-functions.patch
   Upstream: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=a4daaf063f8269a5881154c5b77c5ef6639d65d3

2. target/linux/qualcommax/patches-6.12/0151-arm64-qcom-ipq6018-nss_port5.patch
   Upstream: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=9a7a5d50ee2e035325de9c720e4842d6759d2374

3. target/linux/realtek/patches-6.12/020-01-v6.18-timer-rtl-otto-work-around-dying-timers.patch
   Upstream: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=d0e217b33d42bfe52ef7ef447916a23a586e6e5c

4. target/linux/realtek/patches-6.12/020-03-v6.18-timer-rtl-otto-do-not-interfere-with-interrupts.patch
   Upstream: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=8cc561dd9d02f1753ae34dfdd565662828be9a9d

Additional changes:
- Manually adapted bcm27xx patch:
  * 950-0410-media-i2c-adv7180-Add-support-for-V4L2_CID_LINK_FREQ.patch
    Rebased and adjusted for kernel 6.12 to fix context conflicts.
- Synced lantiq DTS (danube.dtsi) with upstream bindings
  to fix DT validation issues on kernel 6.12.
- Manually adapted DTS to match OpenWrt's lantiq DTS layout.

Compile-tested on x86_64
Run-tested on x86_64

Signed-off-by: gongzi miao <miaogongzi0227@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/20777
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

rsalvaterra reviewed (changes requested):
Please make it a single commit. And make sure you link to the respective upstream commit when you remove an upstreamed patch. See here for an example on how to do it right: 2c12942e5eceb01c8f9b7f3d357b939c9cb060aa.

mgz0227 commented:
> The following information must be present in the commit message, not just on the first comment.
> 
> > changelogs: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.58
> > removed 1.target/linux/generic/backport-6.12/612-01-v6.17-net-dsa-tag_brcm-legacy-reorganize-functions.patch 2.target/linux/qualcommax/patches-6.12/0151-arm64-qcom-ipq6018-nss_port5.patch 3.target/linux/realtek/patches-6.12/020-01-v6.18-timer-rtl-otto-work-around-dying-timers.patch 5.target/linux/realtek/patches-6.12/020-03-v6.18-timer-rtl-otto-do-not-interfere-with-interrupts.patch
> > 1.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.58&id=a4daaf063f8269a5881154c5b77c5ef6639d65d3
> 
> The above link points to the correct head (the v6.15.58 tag). However, the following ones don't…
> 
> > 2.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=9a7a5d50ee2e035325de9c720e4842d6759d2374
> > 3.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=d0e217b33d42bfe52ef7ef447916a23a586e6e5c
> > 4.https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8cc561dd9d02f1753ae34dfdd565662828be9a9d
> 
> The above all point to the 6.12.y branch head, this needs to be fixed.
> 
> > Compile-tested on x86_64 Run-tested on x86_64

Thank you for the feedback.  
All changes have now been incorporated into the commit message, including the changelog, removed patch entries, and corrected upstream commit references (now pointing to the v6.12.58 tag).  
Let me know if anything else is needed.

rsalvaterra reviewed (approved):
Looks good!

GoetzGoerisch commented:
Thank you for your effort.
One nit , please case it `kernel:`

You manually adapted, please mention in the changelog.

"target/linux/bcm27xx/patches-6.12/950-0410-media-i2c-adv7180-Add-support-for-V4L2_CID_LINK_FREQ.patch"

leobsky commented on target/linux/bcm27xx/patches-6.12/950-0410-media-i2c-adv7180-Add-support-for-V4L2_CID_LINK_FREQ.patch:
This manually rebase doesn't seem right https://github.com/raspberrypi/linux/commit/2283ceef3ef0edbcd3615c7419f5851fbaae8ed4

GoetzGoerisch commented:
> I have updated the commit subject to use the lowercase prefix `kernel:`

I think you reverted this with the latest push again.

mgz0227 commented:
> > I have updated the commit subject to use the lowercase prefix `kernel:`
> 
> I think you reverted this with the latest push again.

yes you are right,and now I have updated the commit subject to use the lowercase prefix kernel:

hauke commented on target/linux/bcm27xx/patches-6.12/950-0410-media-i2c-adv7180-Add-support-for-V4L2_CID_LINK_FREQ.patch:
```diff
@@ -15,32 +15,15 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
-+static const s64 adv7180_link_freqs[] = {
-+	[INTERLACED_IDX] = 108000000,
-+	[I2P_IDX] = 216000000,
-+};
```
This patch still looks broken to me. 

You removed many parts of the patch, but the upstream changes are touching different parts, see: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/drivers/media/i2c/adv7180.c?id=v6.12.58&id2=v6.12.57

mgz0227 replied:
Thanks for the feedback!

You are right — this was my mistake. I incorrectly dropped multiple hunks
while adapting the patch to 6.12.58.

I have now rechecked the upstream diff you referenced, restored all missing
parts and refreshed the entire bcm27xx patch to match the upstream changes
correctly.

The patch applies cleanly and the driver compiles again.  
Force-pushed the fixed version. Thanks for pointing this out!

GoetzGoerisch reviewed (commented):
run-tested: realtek/rtl838x
commit 30e0afa3ff15add967254e00f91a994af53f90b4

hnyman commented:
Lantiq xway target (at least danube_arcadyan_arv4510pw) has been broken in phase1 buildbot since this PR got merged.

https://buildbot.openwrt.org/images/#/builders/257/builds/407/steps/38/logs/stdio

```
Error: /builder/shared-workdir/build/build_dir/target-mips_24kc_musl/linux-lantiq_xway/linux-6.12.58/arch/mips/boot/dts/lantiq/danube_arcadyan_arv4510pw.dts:150.1-7 Label or path gpios not found
FATAL ERROR: Syntax error parsing input tree
make[4]: *** [Makefile:183: /builder/shared-workdir/build/build_dir/target-mips_24kc_musl/linux-lantiq_xway/image-danube_arcadyan_arv4510pw.dtb] Error 1
```
